e2e_test/e2e_transfer_test.go

```diff
@@ -31,7 +31,189 @@ const (
+				GatewayFee:          (*hexutil.Big)(new(big.Int).SetInt64(oneCelo / 10)),
+				GatewayFeeRecipient: &gateWayFeeRecipient.Address,
+			},
+			expectErr: true,
+		},
+		{
+			name: "AccessListTxType",
+			txArgs: &ethapi.TransactionArgs{
+				To:         &recipient.Address,
+				Value:      (*hexutil.Big)(new(big.Int).SetInt64(oneCelo)),
+				GasPrice:   (*hexutil.Big)(datum.Mul(datum, new(big.Int).SetInt64(4))),
+				AccessList: &types.AccessList{},
+			},
+			expectErr: false,
+		},
+		{
+			name: "DynamicFeeTxType - tip = MaxFeePerGas - BaseFee",
+			txArgs: &ethapi.TransactionArgs{
+				To:                   &recipient.Address,
+				Value:                (*hexutil.Big)(new(big.Int).SetInt64(oneCelo)),
+				MaxFeePerGas:         (*hexutil.Big)(datum.Mul(datum, new(big.Int).SetInt64(4))),
+				MaxPriorityFeePerGas: (*hexutil.Big)(datum.Mul(datum, new(big.Int).SetInt64(4))),
+			},
+			expectErr: false,
+		},
+		{
+			name: "DynamicFeeTxType - tip = MaxPriorityFeePerGas",
+			txArgs: &ethapi.TransactionArgs{
+				To:                   &recipient.Address,
+				Value:                (*hexutil.Big)(new(big.Int).SetInt64(oneCelo)),
+				MaxFeePerGas:         (*hexutil.Big)(datum.Mul(datum, new(big.Int).SetInt64(4))),
+				MaxPriorityFeePerGas: (*hexutil.Big)(datum),
+			},
+			expectErr: false,
+		},
+		{
+			name: "CeloDynamicFeeTxType - gas = MaxFeePerGas - BaseFee",
+			txArgs: &ethapi.TransactionArgs{
+				To:                   &recipient.Address,
+				Value:                (*hexutil.Big)(new(big.Int).SetInt64(oneCelo)),
+				MaxFeePerGas:         (*hexutil.Big)(datum.Mul(datum, new(big.Int).SetInt64(4))),
+				MaxPriorityFeePerGas: (*hexutil.Big)(datum.Mul(datum, new(big.Int).SetInt64(4))),
+				GatewayFee:           (*hexutil.Big)(new(big.Int).SetInt64(oneCelo / 10)),
+				GatewayFeeRecipient:  &gateWayFeeRecipient.Address,
+			},
+			expectErr: true,
+		},
+		{
+			name: "CeloDynamicFeeTxType - MaxPriorityFeePerGas",
+			txArgs: &ethapi.TransactionArgs{
+				To:                   &recipient.Address,
+				Value:                (*hexutil.Big)(new(big.Int).SetInt64(oneCelo)),
+				MaxFeePerGas:         (*hexutil.Big)(datum.Mul(datum, new(big.Int).SetInt64(4))),
+				MaxPriorityFeePerGas: (*hexutil.Big)(datum),
+				GatewayFee:           (*hexutil.Big)(new(big.Int).SetInt64(oneCelo / 10)),
+				GatewayFeeRecipient:  &gateWayFeeRecipient.Address,
+			},
+			expectErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			watcher := test.NewBalanceWatcher(client, []common.Address{sender.Address, recipient.Address, gateWayFeeRecipient.Address, node.Address})
+			blockNum, err := client.BlockNumber(ctx)
+			require.NoError(t, err)
+			signer := types.MakeSigner(devAccounts[0].ChainConfig, new(big.Int).SetUint64(blockNum))
+			tx, err := prepareTransaction(*tc.txArgs, sender.Key, sender.Address, signer, client)
+			require.NoError(t, err)
+			err = client.SendTransaction(ctx, tx)
+			if tc.expectErr {
```

Comment:
Can we check that is the error that we expect and not another?

Reply:
Done